agent/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java

```diff
@@ -0,0 +1,337 @@
+    if (type == byte.class || type == Byte.class) {
+      return data.consumeByte();
+    } else if (type == short.class || type == Short.class) {
+      return data.consumeShort();
+    } else if (type == int.class || type == Integer.class) {
+      return data.consumeInt();
+    } else if (type == long.class || type == Long.class) {
+      return data.consumeLong();
+    } else if (type == float.class || type == Float.class) {
+      return data.consumeFloat();
+    } else if (type == double.class || type == Double.class) {
+      return data.consumeDouble();
+    } else if (type == boolean.class || type == Boolean.class) {
+      return data.consumeBoolean();
+    } else if (type == char.class || type == Character.class) {
+      return data.consumeChar();
+    }
+    // Return null for non-primitive and non-boxed types in ~5% of the cases.
+    // TODO: We might want to return null for boxed types sometimes, but this is complicated by the
+    //       fact that TypeUtils can't distinguish between a primitive type and its wrapper and may
+    //       thus easily cause false-positive NullPointerExceptions.
+    if (!type.isPrimitive() && data.consumeByte((byte) 0, (byte) 19) == 0) {
+      return null;
+    }
+    if (type.isAssignableFrom(String.class)) {
+      return data.consumeString(data.remainingBytes() / 2);
+    } else if (type.isArray()) {
+      if (type == byte[].class) {
+        return data.consumeBytes(data.remainingBytes() / 2);
+      } else if (type == int[].class) {
+        return data.consumeInts(data.remainingBytes() / 2);
+      } else if (type == short[].class) {
+        return data.consumeShorts(data.remainingBytes() / 2);
+      } else if (type == long[].class) {
+        return data.consumeLongs(data.remainingBytes() / 2);
+      } else if (type == boolean[].class) {
+        return data.consumeBooleans(data.remainingBytes() / 2);
+      } else {
+        Object array = Array.newInstance(type.getComponentType(), data.remainingBytes() / 2);
+        for (int i = 0; i < Array.getLength(array); i++) {
+          Array.set(array, i, consume(data, type.getComponentType()));
+        }
+        return array;
+      }
+    } else if (type.isAssignableFrom(ByteArrayInputStream.class)) {
+      return new ByteArrayInputStream(data.consumeBytes(data.remainingBytes() / 2));
+    } else if (type.isEnum()) {
+      return data.pickValue(type.getEnumConstants());
+    } else if (type == Class.class) {
+      return YourAverageJavaClass.class;
+    } else if (type == Method.class) {
+      return data.pickValue(YourAverageJavaClass.class.getMethods());
+    } else if (type == Constructor.class) {
+      return data.pickValue(YourAverageJavaClass.class.getConstructors());
+    } else if (type.isInterface() || Modifier.isAbstract(type.getModifiers())) {
+      List<Class<?>> implementingClasses = implementingClassesCache.get(type);
+      if (implementingClasses == null) {
```

Comment:
I'm wondering if this would work when classes are loaded dynamically during execution